bignumber.js

```diff
@@ -195,7 +195,8 @@
       if (b == null) {
 
         // Duplicate.
-        if (n instanceof BigNumber) {
+        if (n instanceof BigNumber 
+          || (n != null && n.s != null && n.e != null && n.c != null)) {
```

Comment:
BigNumber instances with the value `NaN` or `Infinity` have `null` as a property value, so 

```js
|| (n && typeof n.s !== 'undefined' && typeof n.e !== 'undefined' && typeof n.c !== 'undefined')) {
```